### PR TITLE
Support let (non-mut) decl without initialization, and later init of non-mut let

### LINKED ISSUE #29: Verus demands `let mut` where Rust does not
If you create a variable and leave off the initializer, and declare it later,

```
let x: type;
// ...
x = ...;
```

then Rust doesn't require a `mut` on the declaration. However, Verus currently requires it (see the `"variable must be declared 'mut' to allow assignment"` error in `vir/src/modes.rs`)

### PR DESCRIPTION
**Pull request by utaal**
Fixes #29

### CODE REVIEW

**Comment by Chris-Hawblitzel** [comment]
What about this:

```
fn test() {
    #[spec] let x: int;
    x = 1;
    x = 2;
    assert(false);
}
```

File: source/vir/src/ast.rs [hunk 1]
```diff
@@ -315,7 +315,7 @@ pub enum ExprX {
     /// Choose specification values satisfying a condition, compute body
     Choose { params: Binders<Typ>, cond: Expr, body: Expr },
     /// Assign to local variable
-    Assign(Expr, Expr),
+    Assign { is_init: bool, lhs: Expr, rhs: Expr }, // (is_init, lhs, rhs)
```
Comment (tjhance) [inline source/vir/src/ast.rs:318]: i think you can remove this comment here

**Comment by utaal** [comment]
> What about this:
> 
> ```
> fn test() {
>     #[spec] let x: int;
>     x = 1;
>     x = 2;
>     assert(false);
> }
> ```

---

I forgot about this, thank you. It should now be fixed (by disallowing delayed initialization for spec variables).

**Review by Chris-Hawblitzel — APPROVED** [review]
Minor comment: `init_non_mut` might be clearer than `not_mut_init` to emphasize that it is an initialization

**Pull request merged**